File: invesdwin-context-parent/invesdwin-context-scala/src/test/java/de/invesdwin/context/scala/tests/InputsAndResultsTestLong.java
package de.invesdwin.context.scala.tests;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import javax.annotation.concurrent.NotThreadSafe;

import org.apache.commons.lang3.ArrayUtils;
import org.springframework.core.io.ClassPathResource;

import de.invesdwin.context.integration.script.IScriptTaskEngine;
import de.invesdwin.context.integration.script.IScriptTaskInputs;
import de.invesdwin.context.integration.script.IScriptTaskResults;
import de.invesdwin.context.scala.AScriptTaskScala;
import de.invesdwin.context.scala.IScriptTaskRunnerScala;
import de.invesdwin.util.assertions.Assertions;

@NotThreadSafe
public class InputsAndResultsTestLong {

    private final IScriptTaskRunnerScala runner;

    public InputsAndResultsTestLong(final IScriptTaskRunnerScala runner) {
        this.runner = runner;
    }

    public void testLong() {
        //putLong
        final long putLong = 123;

        //putLongVector
        final long[] putLongVector = new long[3];
        for (int i = 0; i < putLongVector.length; i++) {
            putLongVector[i] = Long.parseLong((i + 1) + "" + (i + 1));
        }

        //putLongVectorAsList
        final List<Long> putLongVectorAsList = Arrays.asList(ArrayUtils.toObject(putLongVector));

        //putLongMatrix
        final long[][] putLongMatrix = new long[4][];
        for (int row = 0; row < putLongMatrix.length; row++) {
            final long[] vector = new long[3];
            for (int col = 0; col < vector.length; col++) {
                vector[col] = Long.parseLong((row + 1) + "" + (col + 1));
            }
            putLongMatrix[row] = vector;
        }

        //putLongMatrixAsList
        final List<List<Long>> putLongMatrixAsList = new ArrayList<List<Long>>(putLongMatrix.length);
        for (final long[] vector : putLongMatrix) {
            putLongMatrixAsList.add(Arrays.asList(ArrayUtils.toObject(vector)));
        }

        new AScriptTaskScala<Void>() {

            @Override
            public void populateInputs(final IScriptTaskInputs inputs) {
                inputs.putLong("putLong", putLong);

                inputs.putLongVector("putLongVector", putLongVector);

                inputs.putLongVectorAsList("putLongVectorAsList", putLongVectorAsList);

                inputs.putLongMatrix("putLongMatrix", putLongMatrix);

                inputs.putLongMatrixAsList("putLongMatrixAsList", putLongMatrixAsList);
            }

            @Override
            public void executeScript(final IScriptTaskEngine engine) {
                engine.eval(new ClassPathResource(InputsAndResultsTestLong.class.getSimpleName() + ".js",
                        InputsAndResultsTestLong.class));
            }

            @Override
            public Void extractResults(final IScriptTaskResults results) {
                //getLong
                final Long getLong = results.getLong("getLong");
                Assertions.assertThat(putLong).isEqualTo(getLong);

                //getLongVector
                final long[] getLongVector = results.getLongVector("getLongVector");
                Assertions.assertThat(putLongVector).isEqualTo(getLongVector);

                //getLongVectorAsList
                final List<Long> getLongVectorAsList = results.getLongVectorAsList("getLongVectorAsList");
                Assertions.assertThat(putLongVectorAsList).isEqualTo(getLongVectorAsList);

                //getLongMatrix
                final long[][] getLongMatrix = results.getLongMatrix("getLongMatrix");
                Assertions.assertThat(putLongMatrix).isEqualTo(getLongMatrix);

                //getLongMatrixAsList
                final List<List<Long>> getLongMatrixAsList = results.getLongMatrixAsList("getLongMatrixAsList");
                Assertions.assertThat(putLongMatrixAsList).isEqualTo(getLongMatrixAsList);
                return null;
            }
        }.run(runner);
    }

}

File: invesdwin-context-parent/invesdwin-context-scala/src/test/java/de/invesdwin/context/scala/tests/InputsAndResultsTestLong.scala
println("getLong")
if(getLong != null)
	throw new Exception("getLong already defined!")
val getLong = putLong
val getLongType = getLong.getClass()
println(getLongType)
println(getLong)
if(getLongType != classOf[java.lang.Long])
	throw new Exception("getLong not Long!")

println("getLongVector")
if(getLongVector != null)
	throw new Exception("getLongVector already defined!")
val getLongVector = putLongVector.asInstanceOf[Array[Long]]
val getLongVectorType = getLongVector(0).getClass()
println(getLongVectorType)
println(getLongVector)
if(getLongVectorType != classOf[Long])
	throw new Exception("getLongVector not Long!")

println("getLongVectorAsList")
if(getLongVectorAsList != null)
	throw new Exception("getLongVectorAsList already defined!")
val getLongVectorAsList = putLongVectorAsList.asInstanceOf[Array[Long]]
val getLongVectorAsListType = getLongVectorAsList(0).getClass()
println(getLongVectorAsListType)
println(getLongVectorAsList)
if(getLongVectorAsListType != classOf[Long])
	throw new Exception("getLongVectorAsList not Long!")

println("getLongMatrix")
if(getLongMatrix != null)
	throw new Exception("getLongMatrix already defined!")
val getLongMatrix = putLongMatrix.asInstanceOf[Array[Array[Long]]]
val getLongMatrixType = getLongMatrix(0)(0).getClass()
println(getLongMatrixType)
println(getLongMatrix)
if(getLongMatrixType != classOf[Long])
	throw new Exception("getLongMatrix not Long!")

println("getLongMatrixAsList")
if(getLongMatrixAsList != null)
	throw new Exception("getLongMatrixAsList already defined!")
val getLongMatrixAsList = putLongMatrixAsList.asInstanceOf[Array[Array[Long]]]
val getLongMatrixAsListType = getLongMatrixAsList(0)(0).getClass()
println(getLongMatrixAsListType)
println(getLongMatrixAsList)
if(getLongMatrixAsListType != classOf[Long])
	throw new Exception("getLongMatrixAsList not Long!")
